README.md

```diff
@@ -1,14 +1,615 @@
+features:
+
+- Chat rooms for 1:1, 1:many, many:1 and many:many participation.
+- Sending and receiving chat messages.
+- Online status aka presence of chat participants.
+- Chat room occupancy, i.e total number of connections and presence members.
+- Typing indicators
+- Room-level reactions (ephemeral at this stage)
+
+If there are other features you'd like us to prioritize, please [let us know](https://forms.gle/mBw9M53NYuCBLFpMA).
+
+## Usage
+
+You will need the following prerequisites:
+
+- An Ably account
+    - You can [sign up](https://ably.com/signup) to the generous free tier.
+- An Ably API key
+    - Use the default or create a new API key in an app within
+      your [Ably account dashboard](https://ably.com/dashboard).
+    - Make sure your API key has the
+      following [capabilities](https://ably.com/docs/auth/capabilities): `publish`, `subscribe`, `presence`, `history` and
+      `channel-metadata`.
+
+[//]: # (## Installation)
+
+[//]: # ()
+[//]: # (The Ably Chat SDK is available on the Maven Central Repository. To include the dependency in your project, add the following to your `build.gradle` file:)
+
+[//]: # ()
+[//]: # (For Groovy:)
+
+[//]: # ()
+[//]: # (```groovy)
+
+[//]: # (implementation 'com.ably.chat:android')
+
+[//]: # (```)
+
+[//]: # ()
+[//]: # (For Kotlin Script &#40;`build.gradle.kts`&#41;:)
+
+[//]: # ()
+[//]: # (```kotlin)
+
+[//]: # (implementation&#40;"com.ably.chat:android"&#41;)
+
+[//]: # (```)
+
+[//]: # ()
+[//]: # (### Dependency on ably-android)
+
+[//]: # ()
+[//]: # (Certain functionalities are powered by the [ably-android]&#40;https://github.com/ably/ably-java&#41; library. )
+
+[//]: # (The `ably-android` library is included as an api dependency within the Chat SDK, so there is no need to manually add it to your project.)
+
+## Versioning
+
+The Ably client library follows [Semantic Versioning](http://semver.org/). See https://github.com/ably/ably-chat-kotlin/tags for a list of
+tagged releases.
+
+## Instantiation and authentication
+
+To instantiate the Chat SDK, create an [Ably client](https://ably.com/docs/getting-started/setup) and pass it into the
+Chat constructor:
+
+```kotlin
+import com.ably.chat.ChatClient
+import io.ably.lib.realtime.AblyRealtime
+import io.ably.lib.types.ClientOptions
+
+val realtimeClient = AblyRealtime(
+    ClientOptions().apply {
+        key = "<api-key>"
+        clientId = "<client-id>"
+    },
+)
+
+val chatClient = ChatClient(realtimeClient)
+```
+
+You can use [basic authentication](https://ably.com/docs/auth/basic) i.e. the API Key directly for testing purposes,
+however it is strongly recommended that you use [token authentication](https://ably.com/docs/auth/token) in production
+environments.
+
+To use Chat you must also set a [`clientId`](https://ably.com/docs/auth/identified-clients) so that clients are
+identifiable. If you are prototyping, you can use `java.util.UUID` to generate an ID.
+
+## Connections
```

Comment:
We should either remove or mark this section as in progress, since we don't have any impl. for the `Connection`.
Pending issue for the same -> https://github.com/ably-labs/ably-chat-kotlin/issues/63

Reply:
Also, external shared realtime instance already provide a way to listen to `connection` events
https://github.com/ably/ably-java?tab=readme-ov-file#connection